deploy/vagrant/core/config_aws.rb

```diff
@@ -25,6 +24,13 @@ def config_aws(config, i, total, name)
       'Name' => TAG + "-" + name,
     }
 	  aws.availability_zone = AVAILABILITY_ZONE
+    if (SUBNET != nil and SUBNET != "")
+      aws.subnet_id = SUBNET
+      aws.associate_public_ip = TRUE
+      aws.security_groups = ENV['AWS_SECURITY_GROUP_ID_TACH']
```

Comment:
https://github.com/mitchellh/vagrant-aws
search for security_groups on the page.

security_groups - An array of security groups for the instance. If this instance will be launched in VPC, this must be a list of security group Name. For a nondefault VPC, you must use security group IDs instead (http://docs.aws.amazon.com/cli/latest/reference/ec2/run-instances.html).